packages/zendesk-adapter/src/filters/custom_objects/custom_object_fields.ts

```diff
@@ -0,0 +1,325 @@
+/*
+*                      Copyright 2023 Salto Labs Ltd.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import {
+  isInstanceElement,
+  Element,
+  InstanceElement,
+  ReferenceExpression, Value, TemplateExpression, isReferenceExpression,
+} from '@salto-io/adapter-api'
+import _ from 'lodash'
+import { references as referencesUtils, client as clientUtils } from '@salto-io/adapter-components'
+import { collections } from '@salto-io/lowerdash'
+import { logger } from '@salto-io/logging'
+import { inspectValue } from '@salto-io/adapter-utils'
+import { FilterCreator } from '../../filter'
+import {
+  CUSTOM_OBJECT_FIELD_OPTIONS_TYPE_NAME,
+  CUSTOM_FIELD_OPTIONS_FIELD_NAME,
+  CUSTOM_OBJECT_FIELD_TYPE_NAME,
+  CUSTOM_OBJECT_TYPE_NAME,
+  TICKET_FIELD_TYPE_NAME,
+  TRIGGER_TYPE_NAME,
+  ZENDESK,
+} from '../../constants'
+import { FETCH_CONFIG } from '../../config'
+import {
+  LOOKUP_REGEX,
+  RELATIONSHIP_FILTER_REGEX,
+  transformCustomObjectLookupField,
+  transformRelationshipFilterField,
+} from './utils'
+import { paginate } from '../../client/pagination'
+import { getIdByEmail } from '../../user_utils'
+
+const { makeArray } = collections.array
+const { createMissingInstance } = referencesUtils
+const { createPaginator } = clientUtils
+const log = logger(module)
+
+const relationTypeToType = (relationshipTargetType: Value): string => {
+  if (!_.isString(relationshipTargetType)) {
+    return 'unknown'
+  }
+
+  if (relationshipTargetType.startsWith('zen:custom_object')) {
+    return CUSTOM_FIELD_OPTIONS_FIELD_NAME
+  }
+  switch (relationshipTargetType) {
+    case 'zen:user':
+      return 'user'
+    case 'zen:organization':
+      return 'organization'
+    case 'zen:ticket':
+      return TICKET_FIELD_TYPE_NAME
+    default:
+      log.warn(`Unknown relationship target type: ${inspectValue(relationshipTargetType)}`)
+      return 'unknown'
+  }
+}
+
+type Rule = {
+  field: string | TemplateExpression
+  operator: string
+  value?: string | ReferenceExpression
+}
+
+const isRelevantRule = (rule: Value): rule is Rule =>
+  _.isPlainObject(rule)
+  && _.isString(rule.operator)
+  && (rule.value === undefined || _.isString(rule.value))
+  && RELATIONSHIP_FILTER_REGEX.test(rule.field)
```

Comment:
please surround with parens when there's a multiline implementation
```suggestion
const isRelevantRule = (rule: Value): rule is Rule => (
  _.isPlainObject(rule)
  && _.isString(rule.operator)
  && (rule.value === undefined || _.isString(rule.value))
  && RELATIONSHIP_FILTER_REGEX.test(rule.field)
)
```